csharp/protos/google/protobuf/field_presence_test.proto

```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+
+package Google.ProtocolBuffers.TestProtos.FieldPresence;
+
+option java_package = "com.google.protobuf";
+option java_outer_classname = "FieldPresenceTestProto";
+option java_generate_equals_and_hash = true;
+
+message TestAllTypes {
+  enum NestedEnum {
+    FOO = 0;
+    BAR = 1;
+    BAZ = 2;
+  }
+  message NestedMessage {
+    optional int32 value = 1;
```

Comment:
Given that this is proto3, I'd remove the "optional" everywhere. (It's at least non-idiomatic.)